Framework/PythonInterface/plugins/algorithms/GenerateLogbook.py

```diff
@@ -278,7 +278,7 @@ def _perform_binary_operations(self, values, binary_operations, operations):
                     new_val = "N/A"
             else:
                 raise RuntimeError("Unknown operation: {}".format(operation))
-            if type(new_val) == str():
+            if type(new_val) is str():
```

Comment:
Current statement will always be False since `str()` will create and empty string instance.
```suggestion
            if type(new_val) is str:
```